docs/adapters/configuration.md

```diff
@@ -0,0 +1,105 @@
+## Goal
+Most simply and consistently across adapters, allow users to configure one or multiple connections in addition to a single path to the adapter module. 
+
```

Comment:
Let's clarify by adding something like this:

When a Juttle program specifies `read adapter` or `write adapter`, a default instance of that adapter will be selected to read from / write to. The default instance is the top one as specified in the juttle config file for that adapter type (or the only one, in case of one configured instance). If there are multiple configured instances of the adapter, the user can select which one to read/write from by specifying the instance id as an option to read/write source, in some way. A few alternative approaches exist.